bin/land.js

```diff
@@ -0,0 +1,205 @@
+#!/usr/bin/env node
+
+'use strict';
+
+const childProcess = require('child_process');
+const https = require('https');
+const path = require('path');
+const util = require('util');
+const clipboardy = require('clipboardy');
+
+const arg = process.argv.slice(2);
+const landedBranch = arg[0];
+const toolName = path.basename(process.argv[1]);
+const toolVersion = require('../package.json').version;
+const exec = util.promisify(childProcess.exec);
+
+const help = `\
+This a Pull Request landing tool that will automatically pick up commits from
+the branch add metadata to them and merge into the specified branch.
+
+Usage: ${toolName} <target-branch> [OPTION]
+       ${toolName} --help
+       ${toolName} --version
+
+Options:
+  --user          GitHub username
+  --token         GitHub access token
+  --remote-name   point to the remote repo
+  --rebase        start interactive rebase of the source branch
+  --autosquash    move commits that begin with squash!/fixup! during rebase
+  --cherry-pick   apply commits from source branch onto <target-branch>
+  --help          print this help message and exit
+  --version       print version and exit
+`;
+
+const runGit = async function(options) {
+  return new Promise((resolve, reject) => {
+    const git = childProcess.spawn('git', options, { stdio: 'inherit' });
+    git.on('close', code => {
+      if (code !== 0) {
+        reject(code);
+      }
+      resolve(code);
+    });
+  });
+};
+
+async function differCommits() {
+  let git;
+  try {
+    git = await exec(`git rev-list --count master..HEAD`);
+  } catch (error) {
+    console.error(error);
+  }
+  return parseInt(git.stdout.trim());
+}
+
+const httpsGet = (commitMessage, options) => {
+  https
+    .get(options, res => {
+      let data = '';
+      res.setEncoding('utf8');
+
+      res.on('data', chunk => {
+        data += chunk;
+      });
+
+      res.on('end', () => {
+        const prUrl = JSON.parse(data).items[0].html_url;
+
+        const extendedCommit = `${commitMessage}\n\nPR-URL: ${prUrl}`;
+        childProcess.exec(
+          `git commit --amend --message='${extendedCommit}'`,
+          err => {
+            if (err) {
+              console.error(err);
+              process.exit(1);
+            }
+          }
+        );
+      });
+    })
+    .on('error', err => {
+      console.error(err);
+      process.exit(1);
+    });
+};
+
+if (arg.includes('--help') || arg.includes('-h')) {
+  console.log(help);
+  process.exit(0);
+} else if (arg.includes('--version')) {
+  console.log(toolVersion);
+  process.exit(0);
+}
+
+(async () => {
```

Comment:
I'd prefer to put all these handlers in a separate `async` function to avoid wrapping them every time. (like here https://github.com/metarhia/tools/pull/5/files#diff-2be9ce238c15de99fa753e250b3b28fdR278)